platform/ios/src/MGLMapView.h

```diff
@@ -1305,21 +1305,29 @@ MGL_EXPORT IB_DESIGNABLE
 @property (nonatomic, copy) NSArray<id <MGLAnnotation>> *selectedAnnotations;
 
 /**
+ TODO: update documentation
+ TODO: consider deprecation, and replacing with a `selectAnnotation:` that uses the proposed delegate
+ methods.
+
  Selects an annotation and displays its callout view.
 
  The `animated` parameter determines whether the map is panned to bring the
  annotation on-screen, specifically:
 
  | `animated` parameter | Effect |
  |------------------|--------|
- | `NO`             | The annotation is selected, and the callout is presented. However the map is not panned to bring the annotation or callout onscreen. The presentation of the callout is animated. |
+ | `NO`             | The annotation is selected, and the callout is presented. However the map is not panned to bring the annotation or callout onscreen.
  | `YES`            | The annotation is selected, and the callout is presented. If the annotation is offscreen *and* is of type `MGLPointAnnotation`, the map is panned so that the annotation and its callout are brought just onscreen. The annotation is *not* centered within the viewport. |
 
  @param annotation The annotation object to select.
- @param animated If `YES`, the annotation and callout view are animated on-screen.
+ @param animated If `YES`, the annotation and callout view are moved on-screen.
 
  @note In versions prior to `4.0.0` selecting an offscreen annotation did not
  change the camera.
+ 
+ @note The `animated` parameter can be considered the same as the value returned from
+ `-[MGLMapViewDelgate mapView:shouldMoveOnscreenWhenSelectingAnnotation:]`.
```

Comment:
As this note acknowledges, the new delegate method is redundant to the existing `animated` parameter for the purposes of programmatic annotation selection. I’m partial to the existing parameter, both for consistency with MapKit and because delegate methods have a way of sneaking up on you. How many times have you found yourself wondering why callouts aren’t showing up, only to discover that a long view controller implementation never implemented `-[MGLMapViewDelegate mapView:annotationCanShowCallout:]`? The `animated` parameter on this method places the decision to show the annotation and the decision to animate that action in the same place (cf. “[clarity at the point of use](https://swift.org/documentation/api-design-guidelines/#clarity-at-the-point-of-use)”).

As for gesture-based annotation selection, I think `MGLAnnotationView.enabled`, `MGLAnnotationImage.enabled`, `-[MGLMapViewDelegate mapView:shapeAnnotationIsEnabled:]`, and `-[MGLMapViewDelegate mapView:annotationCanShowCallout:]` already provide plenty of opportunity for the developer to disable the panning behavior. More on these points in https://github.com/mapbox/mapbox-gl-native/issues/12064#issuecomment-424593838.

Reply:
I can see the benefit of having the separate methods, since there may be cases where users want the map to pan to some annotations, but not others. But we should make it clear which method takes priority when `animated:` conflicts with `-[MGLMapViewDelgate mapView:shouldMoveOnscreenWhenSelectingAnnotation:]`.